File: runtime/CSlib/opaque.cs
namespace bigloo
{
  public sealed class opaque: obj
  {
    private static opaque _nil = new opaque();

    public static opaque nil() {
      return _nil;
    }
   
    public bool equal( opaque  o )
    {
      return (o == this);
    }

    public int hash()
    {
      return GetHashCode();
    }
  }
}

File: runtime/Jlib/opaque.java
package bigloo;

public class opaque extends obj
{
  private static opaque nil = new opaque();
   
  public byte[] identifier;
  
  public static opaque nil() {
     return nil;
  }
   
  public boolean equal( final opaque  o )
  {
    return (o == this);
  }

  public int hash()
  {
    return hashCode();
  }
}
